### PR TITLE
Add conan profiles for CI to source control

### LINKED ISSUE #1036: Put conan profiles for all platforms in source contol
For reproduciblity and to avoid issues like #1032 .

### PR DESCRIPTION
**Pull request by cqc-alec**
Closes #1036 .

### CODE REVIEW

**Comment by trvto** [comment]
Seems like the CI jobs that may be affected are being skipped due to lack of changes in the code

**Comment by trvto** [comment]
Maybe need to add a filter to determine if workflow itself has changed and run the other jobs if it has

**Review by trvto — CHANGES_REQUESTED** [review]
see comments

**Comment by cqc-alec** [comment]
> Seems like some of the CI jobs that may be affected are being skipped due to lack of changes in the code. Including build_and_test and build_libs

True. I did run both the `build_and_test` and `release` workflows on a separate branch to test these changes, but not `build_libs`.

**Comment by cqc-alec** [comment]
> Maybe need to add a filter to determine if workflow itself has changed and run the other jobs if it has, even if tket/doxyfile/pytket haven't changed
> 
> Looks like `valgrind.yml` and `coverage.yml` do this, maybe use the same mechanism in all workflows

Yes, will add that here.

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -39,6 +39,7 @@ jobs:
         filters: |
           tket:
             - '{tket/src/**,tket/include/**,tket/conanfile.py,tket/CMakeLists.txt,tket/cmake/**,tket/test/**,tket/proptest/**}'
+            - '.github/workflows/build_and_test.yml'
           doxyfile:
             - '{tket/Doxyfile}'
           pytket:
```
Comment (trvto) [inline .github/workflows/build_and_test.yml:45]:
```suggestion
          pytket:
            - 'pytket/**'
            - '.github/workflows/build_and_test.yml'
```

Comment (trvto) [inline .github/workflows/build_and_test.yml:46]: also add here (github won't let me suggest it)

Comment (trvto) [reply .github/workflows/build_and_test.yml:46]: I mean add `- '.github/workflows/build_and_test.yml'` to the doxyfile and pytket filters as well. I just don't get github.

**Review by trvto — APPROVED** [review]
👍

**Pull request merged**